runtime/compiler/il/J9DataTypes.hpp

```diff
@@ -251,17 +251,17 @@ class OMR_EXTENSIBLE DataType : public OMR::DataTypeConnector
    static char *getName(TR_RawBCDSignCode s)
       {
       if (s < num_raw_bcd_sign_codes)
-         return _TR_RawBCDSignCodeNames[s];
+         return (char *)_TR_RawBCDSignCodeNames[s];
```

Comment:
Have you tried declaring this `getName` method to return a `const char *` to avoid all these casts?